### PR TITLE
lockfile v2

### PR DESCRIPTION
**Pull request by mkhl**
an ugly attempt at getting started with lockfile v2 support

patching shebangs probably doesn't work: npm install doesn't treat the lockfile as frozen and npm ci removes node_packages before fetching packages, and the hook with it

related to #140

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -9,7 +9,7 @@ echo -e "Running unit tests\n"
 nix-build -A tests --no-build-output --no-out-link --show-trace
 
 echo -e "\n\nRunning integration tests\n"
-$(nix-build  -A tests.integration-tests --no-out-link --show-trace)
+nix-build  -A tests.integration-tests --no-out-link --show-trace
```
Comment (andir) [inline test.sh:12]: This was actually here to *run* the integration tests. The Nix build will just build the scripts that runs the tests. We still pretty much require this.

**Comment by andir** [comment]
@mkhl Thanks for picking this up. I'll try to give it a go in one of the coming days. This might be a first step towards supporting lockfiles v2 (and newer npm?).

I am guessing this will still fail on any package that requires any of our patch hooks, correct?

File: test.sh (hunk 1 above)
Comment (mkhl) [reply test.sh:12]: oooh i see, thanks for explaining

**Comment by hlolli** [comment]
I gave your pr a try, fails with nodejs 16. It's when calling npmlock2nix.build with installPhase that is just an echo. Perhaps when I don't specify buildPhase, it starts doing something strange.

```
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
/nix/store/0qlq4s175mzlx9g0aqcdk95j92g1z7mf-stdenv-darwin/setup: line 1364: $TMP/preinstall-env: ambiguous redirect
```

**Comment by hlolli** [comment]
ok so I have symbols in my derivation name which are being converted to question mark signs, causing question marks in TMP environment variable, causing ambiguous redirect in bash. Not sure why it doesn't happen on master.

**Comment by andir** [comment]
This shouldn't be required anymore as we've just merged #166.

**Pull request closed without merging**